### PR TITLE
Add AWSServices discard_regex integration tests

### PR DESCRIPTION
**Pull request by fdalmaup**
|Related issue|
|-------------|
|     https://github.com/wazuh/wazuh/issues/17388        |

## Description

This PR adds new test cases for the `discard_regex` functionality, more precisely for the `CloudWatchLogs` and `Inspector` services.

#### Tier 0

- `discard_regex` parameter tests for `AWSCloudWatchLogs`
- `discard_regex` parameter tests for `AWSInspector`

## Testing performed

| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @fdalmaup (Developer)  |   `tests/integration/test_aws/test_discard_regex.py`        | ⚫⚫⚫ | :green_circle: |   Ubuntu 22.04      |    e50f7745b5cfa7c7b3af20e6bef4cdf2e3194e71     | Nothing to highlight |
| @user (Reviewer)   |           | ⚫⚫⚫ | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |        |         | Nothing to highlight |

### CODE REVIEW

**Review by EduLeon12 — DISMISSED** [review]
After reviewing the proposed test cases I think they fulfill the development made, Therefore it's approved.

**Review by Selutario — CHANGES_REQUESTED** [review]
Code changes LGTM. However, AWS ITs are available in 4.6.0 so the base branch can be changed. Also, take a look at the failing checks to fix them.

**Review by Selutario — APPROVED** [review]
LGTM

**Comment by Selutario** [comment]
This PR can be merged once this issue is finished:
- https://github.com/wazuh/wazuh/issues/18110

**Review by Selutario — DISMISSED** [review]

**Review by mauromalara — CHANGES_REQUESTED** [review]
I requested some changes but, for me, the most important things to be considered before proceeding with applying those changes are:

- I see that all test functions do the same and change minimal things like configuration or metadata. I would compress everything in a single test with different cases defined in a single YAML so that the test, according to these cases, knows what to validate and what not. **If this is not possible, please detail what and why so that we can discuss about it.**

- I do not see a test case that checks that the JSON field on which you want to filter does not exist (negative case).

- I don't see a validation to test for not discarding logs that don't apply to the filter set in `<discard_regex>` (negative case).

**Comment by mauromalara** [comment]
> The single YAML approach was tested, but due to the differences between the base configuration of the module for each case, these failed to execute as expected (e.g., for Cloudwatch having and not having the field attribute inside the discard_regex parameter depending on if it is JSON or simple text failed because it added an empty value for the --discard-field parameter to the module's execution).
What you mentioned could be analyzed in the https://github.com/wazuh/wazuh/issues/17788.

Great! Please, update the issue you mentioned before to explicitly describe that you need to change this to reduce duplicated code.

> The test environment contains both JSON and simple text logs. If the JSON field does not exist, the module does not show any related message, it processes every available log without discarding them.

In my view, I think this might not be good for the user. Picture this situation: as a user, what if I make a mistake while specifying the JSON field? Wouldn't it be preferable to promptly detect such issues by examining the logs? I suggest we create an issue on this matter and, furthermore, develop a test case to encompass this particular scenario.

> This is also related to the test environment mentioned previously, if more logs than expected were discarded due to unexpected behavior of the module, then the pattern defined in the tests should find more matches than the ones described in the test cases and fail.

You're right, didn't think about it.

File: tests/integration/test_aws/data/test_cases/discard_regex_test_module/cases_cloudwatch_discard_regex_simple_text.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+- name: cloudwatch_discard_regex_simple_text
+  description: CloudWatch discard regex configuration for simple text logs
```
[the remark this answers is not in the document]
Comment (fdalmaup) [reply tests/integration/test_aws/data/test_cases/discard_regex_test_module/cases_cloudwatch_discard_regex_simple_text.yaml]: Done

File: tests/integration/test_aws/test_discard_regex.py [hunk 2]
```diff
@@ -3,36 +3,35 @@
...
-def test_discard_regex(
-    configuration, metadata, load_wazuh_basic_configuration, set_wazuh_configuration, clean_s3_cloudtrail_db,
-    configure_local_internal_options_function, truncate_monitored_files, restart_wazuh_function, file_monitoring,
+@pytest.mark.parametrize('configuration, metadata', zip(t0_configurations, t0_configuration_metadata), ids=t0_case_ids)
+def test_bucket_discard_regex(
+        configuration, metadata, load_wazuh_basic_configuration, set_wazuh_configuration, clean_s3_cloudtrail_db,
+        configure_local_internal_options_function, truncate_monitored_files, restart_wazuh_function, file_monitoring,
 ):
     """
     description: Fetch logs excluding the ones that match with the regex.
```
[the remark this answers is not in the document]
Comment (fdalmaup) [reply tests/integration/test_aws/test_discard_regex.py]: Done

File: tests/integration/test_aws/data/test_cases/discard_regex_test_module/cases_inspector_discard_regex.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+- name: inspector_discard_regex
+  description: Inspector discard regex configurations
```
[the remark this answers is not in the document]
Comment (fdalmaup) [reply tests/integration/test_aws/data/test_cases/discard_regex_test_module/cases_inspector_discard_regex.yaml]: Done

**Review by mauromalara — APPROVED** [review]
LGTM!

**Pull request merged**